docs/new/develop/indexes.md

```diff
@@ -0,0 +1,99 @@
+When you execute your Substreams for the first time, you are reading the data stored in the flat files of the Substreams provider. 
+
+To improve the performance, the data accessed by the Substreams is cached, so that the second or the third time that you run the Substreams, you can read from the cache, thus saving time and money. This behavior is illustrated in the following diagram.
+
+<figure><img src="../../.gitbook/assets/develop/cache-substreams-flow.png" width="100%" /></figure>
+
+This data indexing is done implicitly every time you run a Substreams for the first time, but Substreams also allows you to explicitly specify which specific data you want to index for further performance improvement.
```

Comment:
> This data indexing
I would rather say: " This data _caching_ " (not indexing! even if this term is sometimes used that way... it will create confusion here)

> allows you to explicitly specify which specific data you want to index
I would rather say:
"allows you to create an additional index on top of your data"
or something like this, to emphasize that the indexes are "different" than simply caching the mapper data.